src/test/java/com/elasticbox/jenkins/tests/PullRequestManagerTest.java

```diff
@@ -0,0 +1,279 @@
+        waitForNextBuild(project, NEXTBUILD_TIMEOUT, "ensureBuildTriggeredInProject");
+
+        // Test both projects building this pullRequest and this commit
+
+        LOGGER.fine("Check that the job 1 build is started");
+        new Condition() {
+            public boolean satisfied() {
+                return project.getLastBuild() != null;
+            }
+        }.waitUntilSatisfied(NEXTBUILD_TIMEOUT_LONG);
+
+        FreeStyleBuild projectLastBuild1 = project.getLastBuild();
+        if (projectLastBuild1 == null) {
+            LOGGER.fine("Build 1 is null");
+        } else {
+            LOGGER.fine("The job 1 have triggered build with number " + projectLastBuild1.number);
+        }
+
+        Assert.assertNotNull(MessageFormat.format("Build 1 is not triggered on opening of pull request {0} after {1} minutes",
+                pullRequest.getGHPullRequest().getHtmlUrl(), TimeUnit.SECONDS.toMinutes(NEXTBUILD_TIMEOUT_LONG)),
+                project.getLastBuild() );
+
+
+        LOGGER.fine("Check that the job 2 build is started");
+        new Condition() {
+            public boolean satisfied() {
+                return project2.getLastBuild() != null;
+            }
+        }.waitUntilSatisfied(NEXTBUILD_TIMEOUT);
+
+        FreeStyleBuild projectLastBuild2 = project2.getLastBuild();
+        if (projectLastBuild2 == null) {
+            LOGGER.fine("Build 2 is null");
+        } else {
+            LOGGER.fine("The job 2 have triggered build with number " + projectLastBuild2.number);
+        }
+
+        Assert.assertNotNull(MessageFormat.format("Build 2 is not triggered on opening of pull request {0} after {1} minutes",
+                pullRequest.getGHPullRequest().getHtmlUrl(), TimeUnit.SECONDS.toMinutes(NEXTBUILD_TIMEOUT_LONG + NEXTBUILD_TIMEOUT)),
+                project2.getLastBuild());
+
+        // Tests success. Finishing...
+        LOGGER.fine("Test completed succesfully. Cleaning...");
+
+        // Ensure triggered builds are completed
+        waitForCompletion(project, TimeUnit.MINUTES.toSeconds(3) );
+        Assert.assertFalse(MessageFormat.format("Build of pull request {0} for project {1} is still not complete after 3 minutes",
+                pullRequest.getGHPullRequest().getHtmlUrl(), project.getName()),
+                project.getLastBuild().isBuilding() );
+
+        waitForCompletion(project2, TimeUnit.MINUTES.toSeconds(3) );
+        Assert.assertFalse(MessageFormat.format("Build of pull request {0} for project {1} is still not complete after 3 minutes",
+                pullRequest.getGHPullRequest().getHtmlUrl(), project2.getName()),
+                project2.getLastBuild().isBuilding() );
+
+        LOGGER.fine("Closing instances from project 1");
+        instances.addAll(checkBuild(project, TestUtils.GITHUB_USER));
+
+        LOGGER.fine("Closing the pull request and deleting " + instances.size() + " instances.");
```

Comment:
Maybe this should be in a finally. What do you think?

Reply:
That is my point, in this test you may want to always close the pull request and delete the instances. A try-finally will manage it clear and simple.